### ISSUE TITLE
Save the optimizer status for future retraining

### ISSUE THREAD

**Issue by diegovalenzuelaiturra** [lorenlugosch/pretrain_speech_model#2]
I think it would be useful if the optimizer state was also saved, for a future retraining from a checkpoint 
* `torch.save(optimizer.state_dict(), path_to_optimizer)`

**Comment by lorenlugosch** [comment]
Good call---I'll work on adding that. Also, when you restart training, it loads the previous state but starts at epoch 1, so that needs to be fixed too.
